pkgs/development/libraries/science/math/cudnn/extension.nix

```diff
@@ -11,29 +11,23 @@
 final: prev: let
   inherit (final) callPackage;
   inherit (prev) cudaVersion;
-  inherit (prev.lib) attrsets lists versions strings trivial;
-
-  # Utilities
-  # majorMinorPatch :: String -> String
-  majorMinorPatch = (trivial.flip trivial.pipe) [
-    (versions.splitVersion)
-    (lists.take 3)
-    (strings.concatStringsSep ".")
-  ];
+  inherit (prev.lib) attrsets lists versions;
+  inherit (prev.lib.strings) replaceStrings versionAtLeast versionOlder;
```

Comment:
Aren't `versionAtLeast` and `versionOlder` from `versions` for that matter?

Reply:
They’re actually from `strings`: https://teu5us.github.io/nix-lib.html#lib.strings.versionolder. The `versions` attribute set is pretty sparse. I guess if we had a type for versions it would make sense to have those functions there, but they’re just strings :(

Reply:
I wondered the same thing! Hopefully the stdlib gets cleaned up eventually :/